src/http.cc

```diff
@@ -1216,7 +1216,7 @@ HttpStateData::readReply(const CommIoCbParams &io)
             return; // wait for Client::noteMoreBodySpaceAvailable()
         }
 
-        if (virginBodyDestination && virginBodyDestination->buf().hasContent()) {
+        if (virginBodyDestination && !virginBodyDestination->buf().hasPotentialSpace()) {
```

Comment:
> Alex: We will come back here after that PR merges.

PR 2056 got merged. Please reevaluate this condition and the corresponding condition in ModXact::virginConsume() condition. As we have [discussed](https://github.com/measurement-factory/squid/pull/291#pullrequestreview-2722073548), the two conditions should probably be the same (or, to be more precise, should match one another). FTP Client code may also have to be adjusted. Please adjust this PR description accordingly as well.

Reply:
>  FTP Client code may also have to be adjusted

FTP Client does not need adjusting because it already has this condition in Client::calcBufferSpaceToReserve(). So if  adaptor_space is zero, calcBufferSpaceToReserve() returns 0 and Ftp::Client::maybeReadVirginBody() exits on the !read_sz condition (i.e., postponing until Client::noteMoreBodySpaceAvailable()). I also ran several tests with FTP requests (configuring delay pools and ICAP as before), and they worked without getting stuck.